src/core/qgsxmlutils.cpp

```diff
@@ -270,3 +270,13 @@ QVariant QgsXmlUtils::readVariant( const QDomElement &element )
     return QVariant();
   }
 }
+
+QgsStableXmlOrder::QgsStableXmlOrder()
+{
+  qSetGlobalQHashSeed( 42 );
```

Comment:
From the qt docs "From Qt 5.10 and onwards, the only allowed values are 0 and -1. Passing the value -1 will reinitialize the global QHash seed to a random value, while the value of 0 is used to request a stable algorithm for C++ primitive types types (like int) and string types (QString, QByteArray)."

Reply:
Good to know. I'll put 1 there, but it doesn't make a difference so far.
There are two things that remain unstable (on a simple project)

 * project properties (maybe the qhasiterator in its write code is the source of the trouble)
 * the projectCrs.spatialrefsys.geographicflag property seems to switch randomly